### DISCUSSION TITLE
Spatie Media Library File Upload Remove Button

### DISCUSSION THREAD

**Discussion by dev3k** [filamentphp/filament#4698] [Help] [answered]
in normal file input, if you click remove without saving the form, the image reference still exists in the database, so if the user navigated to a different page the value still there 

but in SpatieMediaLibraryFileUpload when you click remove 

in vendor/filament/spatie-laravel-media-library-plugin/src/Forms/Components/SpatieMediaLibraryFileUpload.php 

```
$this->deleteUploadedFileUsing(static function (SpatieMediaLibraryFileUpload $component, string $file): void {
            if (! $file) {
                return;
            }
            $mediaClass = config('media-library.media_model', Media::class);
            $mediaClass::findByUuid($file)?->delete(); // this will delete media record
});
```

so without saving/submitting the form you already deleted that image, is there a solution for such behavior to match normal file input?

**Comment by ceejayoz** [comment]
I've filed a bug (#5971) regarding this behavior.

**Comment by danharrin** [comment, marked as answer]
Fixed by #6083.

**Reply by dev3k** [reply]
thanks
